examples/Baselines/Halite_competition/paddle/rl_trainer/agent.py

```diff
@@ -0,0 +1,124 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import parl
+import paddle
+
+
+class Agent(parl.Agent):
+    """Agent.
+    Args:
+        algorithm (`parl.Algorithm`): algorithm to be used in this agent.
+    """
+
+    def __init__(self, algorithm):
+
+        self.alg = algorithm
+
+    def learn(self, obs, act, value, returns, log_prob, adv):
+        """Updating network
+        Args:
+            obs (numpy.array): representation of current observation
+            act (numpy.array): current action
+            value (numpy.array): state value
+            returns (numpy.array): discounted return
+            log_prob (numpy.array): the log probabilities of action
+            adv (numpy.array): advantage value
+        """
+
+        obs = paddle.to_tensor(obs, dtype=paddle.float32)
+        act = paddle.to_tensor(act, dtype=paddle.int32)
+        value = paddle.to_tensor(value, dtype=paddle.float32)
+        returns = paddle.to_tensor(returns, dtype=paddle.float32)
+        log_prob = paddle.to_tensor(log_prob, dtype=paddle.float32)
+        adv = paddle.to_tensor(adv, dtype=paddle.float32)
+
+        value_loss, action_loss, entropy = self.alg.learn(
+            obs, act, value, returns, log_prob, adv)
+
+        return value_loss, action_loss, entropy
+
+    def predict(self, state):
+        """Predict action
+        Args:
+            state (tensor): representation of current state 
```

Comment:
Is the  state `tensor` or `np.array`, for the `state_tensor = paddle.to_tensor(state, dtype=paddle.float32)` used below.